### DISCUSSION TITLE
Support for BetterTouchTool

### DISCUSSION THREAD

**Discussion by ElhemEnohpi** [input-leap/input-leap#1436] [Ideas]
Hi, and thanks for continuing the development of this app!

tl;dr version:
1. Is there a way for an app or script to determine whether the remote screen is active?
2. Can we get Input Leap to recognize virtual keystrokes generated by BetterTouchTool?

I use [BetterTouchTool](https://folivora.ai/) on my Mac for all kinds of trackpad gestures and custom keyboard triggers. It would be great if it would work with Input Leap to control a remote Mac client also running BTT.

Currently, using BTT doesn't work at all. The virtual keystrokes that BTT can generate aren't recognized, and still go to the host rather than the remote. For example, I have a four-finger tap gesture set to generate a command-H keystroke, to hide the current app. But when the cursor is on the remote screen, command-H isn't sent to it - the current app on the host gets hidden instead. This is also true in terms of recognizing a virtual keystroke from BTT as a Hotkey, as defined in the Server Configuration. It would be helpful if these virtual keystrokes could be recognized by Input Leap. Should I open an issue with a feature request?

That would be useful in very simple setups, where all gestures are global and they only trigger keystrokes. But BTT also lets you define gestures on a per-application basis, and trigger many other sorts of actions. One way this could be supported is to configure BTT on the host to send unique keystrokes representing gestures, which would then be interpreted by BTT on the remote. For example, you could set four-finger tap to send control-option-F1, which could be configured on the client to do different actions, depending on the current app. BTT on the host would also be configured to respond to control-option-F1 in the same way (BTT can send a keystroke from a trackpad gesture, then pick that up itself as a keyboard shortcut to trigger another keystroke or action).

Some people have worked around this by [using BTT's built-in web server to send gesture data](https://gist.github.com/findmory/82593b407ef436fe8a39f1bbb6802690) (I haven't tried it myself). However, because Input Leap works in the background, this requires remembering to use a modifier key with each gesture, in order to signal that it should be sent to the remote. A better option would be for BTT to be aware of when the remote screen is active, and to automatically change the gestures. This could be accomplished with an option to bring Input Leap to the foreground on the server whenever the remote screen is active. If there was a signal that another app could monitor, like an Apple Event on the Mac, a script could bring Input Leap to the foreground, or a "dummy" app with no windows, and the necessary gestures could be configured for that. Possibly BTT itself could monitor that signal, and switch to another set of gestures when the remote is active. There may be various other uses for such a signal. See also: https://github.com/debauchee/barrier/discussions/1513

It's a lot of work for the user to set up duplicate actions for every gesture, so that they can be sent either as unique keystrokes, or as triggers to the web server. It would be nice if BTT could automatically take care of recognizing the gestures on the host, knowing whether Input Leap is active, and if so, sending a command to its web server on the client, then interpreting the gestures there as though they were local. I made a [comment on the BTT forum](https://community.folivora.ai/t/can-btt-detect-mouse-cursor-moved-to-different-monitor-on-another-computer-via-synergy/22024/3) regarding that.
